metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/dao/DataEvolutionUtils.java

```diff
@@ -517,6 +355,81 @@ private static boolean pluginHasSuccessfulRecords(
+   * includes all incremental published harvests (if any) that came after that.
+   * </p>
+   * <p>
+   * Note that a published harvest is a harvest that resulted in a index to publish (i.e. any
+   * harvest that is the root predecessor of an index to publish).
+   * </p>
+   *
+   * @param datasetId The dataset ID for which to obtain the chain.
+   * @return The chain, in the form of plugin-execution pairs that are ordered chronologically. Is
+   * never null, but can be empty if no such chain exists (i.e. the dataset does not have a
+   * published harvest or we find an index after the last full harvest that is invalid or did
+   * somehow not originate from a harvest).
+   */
+  public List<PluginWithExecutionId<? extends ExecutablePlugin<?>>> getPublishedHarvestIncrements(
+          String datasetId) {
+
+    // Get all workflows with finished publish plugins (theoretically we can't quite rely on order).
+    final Pagination pagination = workflowExecutionDao.createPagination(0, null, true);
+    final ResultList<ExecutionDatasetPair> executionsWithPublishOperations = workflowExecutionDao
+            .getWorkflowExecutionsOverview(Set.of(datasetId), Set.of(PluginStatus.FINISHED)
+                    , Set.of(PluginType.PUBLISH), null, null, pagination);
+
+    // Extract all (finished) publish plugins inversely sorted by started date (most recent first).
+    final List<Pair<IndexToPublishPlugin, WorkflowExecution>> publishOperations = new ArrayList<>();
+    executionsWithPublishOperations.getResults().stream().map(ExecutionDatasetPair::getExecution)
+            .forEach(execution -> execution.getMetisPlugins().stream()
+                    .filter(plugin -> plugin instanceof IndexToPublishPlugin)
+                    .map(plugin -> (IndexToPublishPlugin) plugin)
+                    .forEach(plugin -> publishOperations.add(new ImmutablePair<>(plugin, execution))));
+    final Comparator<Pair<? extends ExecutablePlugin<?>, WorkflowExecution>> comparator = Comparator
+            .comparing(pair -> pair.getLeft().getStartedDate());
+    publishOperations.sort(comparator.reversed());
+
+    // Compile a list of all associated harvests (that led to one of these publish operations).
+    // Note: we assume that workflows don't cross each other (i.e. an earlier publish cannot have a
+    // later harvest). We stop when we find a full harvest (the latest full harvest).
+    final Map<ExecutedMetisPluginId, PluginWithExecutionId<? extends ExecutablePlugin<?>>> resultHarvests = new HashMap<>();
+    for (Pair<IndexToPublishPlugin, WorkflowExecution> publishOperation : publishOperations) {
+
+      // If the publish is not available, we have detected an anomaly. We are done.
+      if (MetisPlugin.getDataStatus(publishOperation.getLeft()) != DataStatus.VALID &&
+              MetisPlugin.getDataStatus(publishOperation.getLeft()) != DataStatus.DEPRECATED) {
```

Comment:
So DEPRECATED plugins are ok?
Shouldn't this be allowed for only VALID ones?

Reply:
Yes exactly so why would we allow an incremental harvest to happen if the datase status for the plugin execution is deprecated? I understand this method returns the chain of harvests but, it is used with the purpose to identify if an incremental harvest is allowed or not here eu.europeana.metis.core.dao.WorkflowValidationUtils#isIncrementalHarvestingAllowed
Or I'm I understanding this wrongly?

Reply:
Imagine this sequence of events:
1. Full harvest followed by index
2. Incremental harvest followed by index
3. Reindex (no longer really belongs to either of the previous index operations, but 'deprecates' the last index)
4. Incremental harvest

This last incremental harvest SHOULD be allowed, right? There is no reason to stop an incremental harvest from happening after a reindex. We know that the exact same records are published as before (with possibly different properties), so we should just be able to go ahead with an incremental harvest here.
If we don't allow DEPRECATED plugins, we have a problem here.

Reply:
I see you are right we should allow this!
I think in my mind was more executions other than publish that were deprecated during reindex and that this method would allow indexing without re-running those(perhaps for previous increments, if those previous revision increments were used in combination with the "new" ones), but I guess that doesn't apply..